### PR TITLE
Corrected behaviour if "mod_rewrite" is not enabled.

### PR DESCRIPTION
**Pull request by cmfcmf**
| Q                 | A
| ----------------- | ---
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Tests pass?       | ---
| Fixed tickets     | ---
| Refs tickets      | :de: http://support.zikula.de/module-CMS_Support_Forum-viewtopic-topic-31411-start-0.htm#pid149344
| License           | MIT
| Doc PR            | ---
| Changelog updated | no

### CODE REVIEW

File: src/system/Zikula/Module/SettingsModule/SettingsModuleInstaller.php [hunk 1]
```diff
@@ -74,7 +74,12 @@ public function install()
         System::setVar('shorturls', false);
         System::setVar('shorturlstype', '0');
         System::setVar('shorturlsseparator', '-');
-        System::setVar('shorturlsstripentrypoint', true);
+
+        if (function_exists('apache_get_modules') && in_array('mod_rewrite', apache_get_modules())) {
+            // Only strip entry point if "mod_rewrite" is available.
+            System::setVar('shorturlsstripentrypoint', true);
+        }
+
```
Comment (Guite) [inline src/system/Zikula/Module/SettingsModule/SettingsModuleInstaller.php]: @craigh something to consider in your new installer as well

Comment (cmfcmf) [inline src/system/Zikula/Module/SettingsModule/SettingsModuleInstaller.php]: It's a bit more difficult regarding the installer. I will probably take care of it once the Intaller is finished.

**Comment by craigh** [comment]
so what happens if mod rewrite isn't enabled?

**Comment by Kaik** [comment]
I guess #2057 and it did not work last time when I installed zikula (before this merge) I will check if this is solving the issue.

**Comment by cmfcmf** [comment]
> so what happens if mod rewrite isn't enabled?

Bad stuff :boat: 
1. (fixed by this PR) if stripentrypoint is true, the url will never include `index.php`. Furthermore, the .htaccess rewrite rules aren't applied. This causes all Routes to fail (i.e. `example.com/admin/foo` would search for a file called `foo` in a folder called `admin`, which obviously doesn't exist and display 404).
2. If the URL is `http://example.com` (note the missing `index.php`), the startpage will load, but all routes + debug toolbar fail, because it generates routes and links like `example.com/admin/foo` (without `index.php`). Pending fix: Add rules to the .htaccess (to be done in the installer [I will help with that])

**Comment by craigh** [comment]
would it be prudent to *require* mod_rewrite? Or can we (symfony & routes, etc) get by without it?

**Comment by cmfcmf** [comment]
> would it be prudent to require mod_rewrite?

I also thought about that. I don't have enough experience on whether this is always (or in most of the cases) enabled at hosting providers.

FYI: Drupal 8 requires it as well.

**Pull request merged**